src/hooks/useLensProtocolName.ts

```diff
@@ -0,0 +1,36 @@
+import { ApolloClient, gql, InMemoryCache } from "@apollo/client"
+import { useWeb3React } from "@web3-react/core"
+import useSWRImmutable from "swr/immutable"
+
+const apolloClient = new ApolloClient({
+  uri: "https://api.lens.dev/",
+  cache: new InMemoryCache(),
+})
+
+const fetchLensProtocolName = (_, query) =>
+  apolloClient
+    .query({
+      query: gql(query),
+    })
+    .then((res) => res.data.profiles.items[0].handle)
+
+const useLensProtocolName = () => {
+  const { account } = useWeb3React()
+  const query = `
```

Comment:
Minor impro: you could construct this query inside `fetchLensProtocolName` and pass only the `account` to that function, since that's the only variable data in it.